### ISSUE TITLE
False positive: 'typeof SvelteComponent' is not callable for dynamic components

### ISSUE THREAD

**Issue by janosh** [pheuter/svelte-check-rs#88]
When using `import.meta.glob` to dynamically import Svelte components and then rendering them in an `{#each}` block, svelte-check-rs reports a false positive error about `SvelteComponent` not being callable.

## Reproduction

```svelte
<script lang="ts">
  import type { SvelteComponent } from 'svelte'

  // Simulating import.meta.glob result
  const plots = import.meta.glob(`./plots/*.svelte`, {
    eager: true,
    import: `default`,
  }) as Record<string, typeof SvelteComponent>
</script>

{#each Object.entries(plots) as [name, Plot]}
  <Plot {name} />
{/each}
```

## Error

```
Error: Value of type 'typeof SvelteComponent' is not callable. Did you mean to include 'new'? (ts(TS2348))
```

## Workaround

Use `Component` type from Svelte instead of `typeof SvelteComponent`:

```ts
import type { Component } from 'svelte'
const plots = {} as Record<string, Component>
```

svelte-check-rs: 0.7.1
